### PR TITLE
add workflow_dump + test

### LINKED ISSUE #794: In `ApplyWorkflow`, store a copy of the `Workflow` at the time of submission
This is because the workflow may change during execution.

### PR DESCRIPTION
**Pull request by mfranzon**
closes #794 
remember to update `common.schemas.applyworkflow.ApplyworkflowRead`

### CODE REVIEW

File: fractal_server/app/api/v1/project.py [hunk 1]
```diff
@@ -320,6 +320,10 @@ async def apply_workflow(
         input_dataset_id=input_dataset_id,
         output_dataset_id=output_dataset_id,
         workflow_id=workflow_id,
+        workflow_dump=dict(
+            workflow.dict(),
```
Comment (tcompa) [inline fractal_server/app/api/v1/project.py:324]: Can we exclude `task_list` explicitly, as in `workflow.dict(exclude={"task_list"})`? It'd be to make it a bit more transparent.

File: fractal_server/app/api/v1/project.py [hunk 2]
```diff
@@ -320,6 +320,10 @@ async def apply_workflow(
...
+            task_list=[wf_task.dict() for wf_task in workflow.task_list],
```
Comment (tcompa) [inline fractal_server/app/api/v1/project.py:325]: I think this ignores `wf_task.task`, which would not be OK.

File: fractal_server/app/models/job.py [hunk 3]
```diff
@@ -103,6 +108,8 @@ class ApplyWorkflow(_ApplyWorkflowBase, SQLModel, table=True):
     )
     workflow: Workflow = Relationship()
 
+    workflow_dump: dict[str, Any] = Field(sa_column=Column(JSON))
```
Comment (tcompa) [inline fractal_server/app/models/job.py:111]: Reminder: can you add the description to the docstring?

File: fractal_server/app/models/job.py [hunk 4]
```diff
@@ -79,6 +81,9 @@ class ApplyWorkflow(_ApplyWorkflowBase, SQLModel, table=True):
             (Mapper attribute)
     """
 
+    class Config:
```
Comment (tcompa) [inline fractal_server/app/models/job.py:84]: I'm always confused by this. Is it because of `Column(JSON)`?

**Comment by tcompa** [comment]
Reminders:
* Update CHANGELOG
* Fix https://github.com/fractal-analytics-platform/fractal-common/issues/84 and pull the update common in this branch

File: fractal_server/app/models/job.py (hunk 4 above)
Comment (mfranzon) [reply fractal_server/app/models/job.py:84]: exactly, not a standard type for sqlalchemy https://docs.sqlalchemy.org/en/20/core/type_basics.html#generic-camelcase-types

File: fractal_server/app/api/v1/project.py (hunk 2 above)
Comment (mfranzon) [reply fractal_server/app/api/v1/project.py:325]: true, I'll add in the next commit

File: fractal_server/app/models/job.py (hunk 4 above)
Comment (tcompa) [reply fractal_server/app/models/job.py:84]: Thanks!

File: fractal_server/app/api/v1/project.py (hunk 1 above)
Comment (mfranzon) [reply fractal_server/app/api/v1/project.py:324]: ok

File: fractal_server/app/models/job.py (hunk 3 above)
Comment (mfranzon) [reply fractal_server/app/models/job.py:111]: sure

File: fractal_server/app/api/v1/project.py [hunk 5]
```diff
@@ -320,6 +320,13 @@ async def apply_workflow(
         input_dataset_id=input_dataset_id,
         output_dataset_id=output_dataset_id,
         workflow_id=workflow_id,
+        workflow_dump=dict(
+            workflow.dict(exclude={"task_list"}),
+            task_list=[
+                dict(wf_task.task.dict(), task=wf_task.dict())
```
Comment (tcompa) [inline fractal_server/app/api/v1/project.py:326]: Same as above: what about excluding `task`? Maybe it's a bit pedantic, but it's to make it clear that it won't be duplicated.

Comment (mfranzon) [reply fractal_server/app/api/v1/project.py]: :+1:

**Pull request merged**